carma_cooperative_perception/src/host_vehicle_filter_component.cpp

```diff
@@ -184,10 +190,7 @@ auto euclidean_distance_squared(
   const geometry_msgs::msg::Pose & a, const geometry_msgs::msg::Pose & b) -> double
 {
   return std::pow(a.position.x - b.position.x, 2) + std::pow(a.position.y - b.position.y, 2) +
-         std::pow(a.position.z - b.position.z, 2) + std::pow(a.orientation.x - b.orientation.x, 2) +
-         std::pow(a.orientation.y - b.orientation.y, 2) +
-         std::pow(a.orientation.z - b.orientation.z, 2) +
-         std::pow(a.orientation.w - b.orientation.w, 2);
+         std::pow(a.position.z - b.position.z, 2);
```

Comment:
Should we just take off z component as well since carma doesn't use this component?

Reply:
I don't think so because Euclidean (without any other qualifiers) is typically 3D. If we reduce it to 2D, we should rename the function to `pythagorean_distance_squared()` or something similar.

Since CARMA does not use the z-axis, this term will be zero anyways.